File: rb/features.rb
# FreeBirds SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module FreeBirdsFeatures
  def self.make_feature(name)
    case name
    when "base"
      FreeBirdsBaseFeature.new
    when "test"
      FreeBirdsTestFeature.new
    else
      FreeBirdsBaseFeature.new
    end
  end
end

File: py/features.py
# FreeBirds SDK feature factory

from feature.base_feature import FreeBirdsBaseFeature
from feature.test_feature import FreeBirdsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: FreeBirdsBaseFeature(),
        "test": lambda: FreeBirdsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
